code/implementation/host-stats/src/main/java/io/cattle/platform/host/stats/api/HostStatsOutputFilter.java

```diff
@@ -16,15 +20,38 @@
     @Override
     public Resource filter(ApiRequest request, Object original, Resource converted) {
         boolean add = false;
+        boolean hostStats = false;
+        boolean containerStats = false;
+        boolean project = false;
 
         if (original instanceof Host && DockerHostConstants.KIND_DOCKER.equals(((Host) original).getKind())) {
             add = true;
+            hostStats = true;
+            containerStats = true;
         } else if (original instanceof Instance && InstanceConstants.KIND_CONTAINER.equals(((Instance) original).getKind())) {
+            containerStats = true;
             add = true;
         }
 
+        if (original instanceof Account && AccountConstants.PROJECT_KIND.equals(((Account) original).getKind())) {
+            project = true;
+        }
+        if (original instanceof Service) {
+            containerStats = true;
+        }
+
         if (add) {
-            converted.getLinks().put(HostStatsConstants.LINK_STATS, ApiContext.getUrlBuilder().resourceLink(converted, HostStatsConstants.LINK_STATS));
+            converted.getLinks().put(StatsConstants.LINK_STATS, ApiContext.getUrlBuilder().resourceLink(converted, StatsConstants.LINK_STATS));
+        }
+        if (hostStats) {
+            converted.getLinks().put(StatsConstants.HOST_STATS, ApiContext.getUrlBuilder().resourceLink(converted, StatsConstants.HOST_STATS));
+        }
+        if (containerStats) {
```

Comment:
Minor: if these conditions are mutually exclusive, it would probably make sense to do elseif so the next condition is not checked if the previous one is evaluated to true.